Example/Example/InceptionExample.swift

```diff
@@ -0,0 +1,162 @@
+    var model: mobilenet!
+
+    var selectedIndex = 0
+    var commandQueue: MTLCommandQueue!
+
+    var textureLoader: MTKTextureLoader!
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var label: UILabel!
+    var classLabels: [String]?
+    var inceptionLabels: [String]?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.commandQueue = Device.shared.makeCommandQueue()
+        textureLoader = MTKTextureLoader(device: Device.shared)
+        setupNetwork()
+        classLabels = readClassLabels("mobilenet_labels")
+        inceptionLabels = readClassLabels("imagenet_label_strings")
+        model = mobilenet()
+    }
+
+    func setupNetwork() {
+        guard let mobilenetUrl = Bundle.main.url(forResource: "mobilenet_frozen", withExtension: "pb") else {
+            return
+        }
+        let converter = TFConverter.default()
+
+        mobilenetwork = Network.load(url: mobilenetUrl, inputSize: LayerSize(h: 224, w: 224, f: 3), converter: converter, performInitialize: false)
+
+        // after adding all our layers we are able to initialize the network
+        mobilenetwork.addPreProcessing(layers: [Neuron(type: ActivationNeuronType.custom(neuron: MPSCNNNeuronLinear(device: Device.shared, a: 2.0, b: -1)), id: "scale_neuron")])
+        mobilenetwork.initialize()
+
+        guard let inceptionUrl = Bundle.main.url(forResource: "inception_v3", withExtension: "pb") else {
+            print("Download Inception from: http://download.tensorflow.org/models/inception_v3_2016_08_28.tar.gz" +
+                  " and freeze it using Benderthon or download a frozen .pb file. Leave it as 'inception_v3.pb' in the Example/data folder")
+            return
+        }
+        inception = Network.load(url: inceptionUrl, inputSize: LayerSize(h: 299, w: 299, f: 3), converter: converter, performInitialize: false)
+
+        // after adding all our layers we are able to initialize the network
+        inception?.addPreProcessing(layers: [Neuron(type: ActivationNeuronType.custom(neuron: MPSCNNNeuronLinear(device: Device.shared, a: 2.0, b: -1)))])
+        inception?.initialize()
+    }
+
+    @IBAction func runNetwork(_ sender: Any) {
+        if selectedIndex == 0 {
+            guard let image = imageView.image?.toMPS(loader: textureLoader) else {
+                return
+            }
+            mobilenetwork.run(input: image, queue: commandQueue) { [weak self] result in
+                let arr = result.toArray()
+                let top5 = arr.argsort(by: > ).prefix(5)
+                self?.label.text = top5.reduce("") {
+                    return $0 + (self?.classLabels?[$1] ?? "__") + "\n"
+                }
+            }
+        } else if selectedIndex == 2 {
+            guard let inception = inception, let image = imageView.image?.toMPS(loader: textureLoader) else {
+                return
+            }
+            inception.run(input: image, queue: commandQueue) { [weak self] result in
```

Comment:
maybe encapsulate this repeated logic of getting the image from the imageView and running the network